src/isomorphic/classic/element/ReactElement.js

```diff
@@ -208,6 +208,12 @@ ReactElement.createElement = function(type, config, children) {
         'React.createElement(...): Expected props argument to be a plain object. ' +
         'Properties defined in its prototype chain will be ignored.'
       );
+
+      warning(
```

Comment:
@zemlanin would you be able to verify that there are no significant performance regressions in DEV with this change?